pandas/tests/groupby/test_categorical.py

```diff
@@ -1246,6 +1253,16 @@ def test_get_nonexistent_category():
     # Accessing a Category that is not in the dataframe
     df = pd.DataFrame({"var": ["a", "a", "b", "b"], "val": range(4)})
     with pytest.raises(KeyError, match="'vau'"):
+        df.groupby("var").apply(
+            lambda rows: pd.DataFrame({"val": [rows.iloc[-1]["vau"]]})
```

Comment:
nitpick: if its the `.apply` that raises, let's do the df.groupby outside of the pytest.raises block, e.g. `gb = df.groupby("var")`